### DISCUSSION TITLE
[node]@14 defines type `Sign`, but definitions call it `Signer`

### DISCUSSION THREAD

**Discussion by cjbarth** [DefinitelyTyped/DefinitelyTyped#65740] [Issues with a @types package] [answered]
In Node 14, there is a return type of `Sign` for `crypto.createSign()` as per the manual [here](https://nodejs.org/docs/latest-v14.x/api/all.html#crypto_crypto_createsign_algorithm_options). However, the [definitions](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a578a934d741796e02f961f6303dec3dc9b6e2a5/types/node/v14/crypto.d.ts#L337) say that it returns `Signer`. This differs from the same [definitions](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a578a934d741796e02f961f6303dec3dc9b6e2a5/types/node/v18/crypto.d.ts#L1292) for Node 18 which say the return type is `Sign`, which matches the manual for Node 14 and Node 18. The Node 18 [manual](https://nodejs.org/dist/latest-v18.x/docs/api/all.html#all_crypto_cryptocreatesignalgorithm-options) hasn't changed from the Node 14 manual in this matter: it has always returned `Sign`, so I expect the definitions would be the same too.

I'd put up a PR for this, but since it is Node, I wanted to get the community feel on this one first since it is a rather big change.

[a comment by an automated account is not shown]

**Comment by Semigradsky** [comment]
PR is welcome!

**Reply by cjbarth** [reply, marked as answer]
Please find the PR [here](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/65743).
